app/grandchallenge/reader_studies/tasks.py

```diff
@@ -11,6 +11,13 @@ def add_score(obj, answer):
     obj.save()
 
 
+@transaction.atomic
+def add_image(obj, image):
+    obj.answer_image = image
+    obj.save()
+    image.update_answer_image_viewer_permissions()
```

Comment:
Are the `"view_image"` permissions updated if an image is replaced?

Reply:
With this I'm just wondering if we should remove these permissions for the creator of the answer if the image is changed? I think it would be ok for them to be able to view the image in any case. That would make assigning the permissions a bit more straight forward as well, as otherwise we'd have to check for users that have view permissions and are not in any of the expected groups.

Reply:
I'm not sure that I understand, could you clarify this?

Reply:
Sorry, I did not realize I hadn't pushed the changes I made yet, hopefully that will make things a bit more clear.

What I meant was that if the image for an answer is changed, I think we should still leave the view permission intact for the user that uploaded the old image. I do not see any issue with doing so. The last part of what I wrote refers to the checks performed in this function: https://github.com/comic/grand-challenge.org/pull/1657/files#diff-4776f6f9ffb86c949abe083c2c7613f3991cfdf9387fe11202575e29cbcbba9fR456 (which I have just pushed the changes for). These checks would become needlessly complicated if we do want to remove the permission I think. Because we would then also have to check for users that have viewing permissions but are not in any of the groups that should have them.

Does that make things a bit more clear?